### PR TITLE
verdictd/opa: add verdictd OPA sub-module upper API

### PR DESCRIPTION
**Pull request by EmisonLu**

### CODE REVIEW

File: verdictd/app/src/PolicyEngine/opa/.gitignore [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Binaries for programs and plugins
```
Comment (liangzhou121) [inline verdictd/app/src/PolicyEngine/opa/.gitignore:1]: I think this  .gitignore file was auto generated and shouldn't commit to repro.

File: verdictd/app/src/PolicyEngine/opa/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+[package]
+name = "opa"
+version = "0.1.0"
+authors = ["Emison Lu"]
```
Comment (liangzhou121) [inline verdictd/app/src/PolicyEngine/opa/Cargo.toml:4]: It's better to add your mail info.

**Comment by jiazhang0** [comment]
@EmisonLu Please change the commit header prefix from "opa" to "verdictd/opa".

File: verdictd/app/src/PolicyEngine/opa/opaEngine.rs [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+            println!("Json unmashall failed");
+            return false;
+        }
+    };
+
+    // Generate policy file from reference value
+    let mrEnclave: String = String::from("mrEnclave = ") + &references["mrEnclave"].to_string();
+    let mrSigner: String = String::from("mrSigner = ") + &references["mrSigner"].to_string();
+    let productId: String = String::from("productId = ") + &references["productId"].to_string();
+    let policy = "package demo\n\n".to_string()
```
Comment (liangzhou121) [inline verdictd/app/src/PolicyEngine/opa/opaEngine.rs:47]: I think it's better to remove demo description.

File: verdictd/app/src/PolicyEngine/opa/opaEngine.rs [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+    let mut file = match File::create(&path) {
+        Err(_) => {
+            println!("Couldn't create file");
+            return false;
+        }
+        Ok(file) => file,
+    };
+
+    // Write the policy into file
+    match file.write_all(policy.as_bytes()) {
```
Comment (liangzhou121) [inline verdictd/app/src/PolicyEngine/opa/opaEngine.rs:193]: Need to add a global read-write lock before write files.

File: verdictd/app/src/PolicyEngine/opa/opaEngine.rs (hunk 3 above)
Comment (EmisonLu) [reply verdictd/app/src/PolicyEngine/opa/opaEngine.rs:47]: Do you mean to remove the "demo" information in "let policy = \"package demo\n\n\".to_string()"？

File: verdictd/app/src/PolicyEngine/opa/opaEngine.rs (hunk 4 above)
Comment (EmisonLu) [reply verdictd/app/src/PolicyEngine/opa/opaEngine.rs]: I have added a file lock, please review.

**Review by liangzhou121 — APPROVED** [review]

**Pull request merged**